src/UrqlClient.re

```diff
@@ -1,97 +1,40 @@
-/* Model fetchOptions as a variant, and unwrap the user supplied prop to an object
-   to pass directly to urql. */
+type t;
 type fetchOptions =
   | FetchObj(Fetch.requestInit)
   | FetchFn(unit => Fetch.requestInit);
 
-let unwrapFetchOptions: fetchOptions => Fetch.requestInit =
-  (opts: fetchOptions) =>
-    switch (opts) {
-    | FetchObj(obj) => obj
-    | FetchFn(fn) => fn()
-    };
-
-type data;
-type write = (~hash: string, ~data: data) => Js.Promise.t(unit);
-type read('value) = (~hash: string) => Js.Promise.t('value);
-type invalidate = (~hash: string) => Js.Promise.t(unit);
-type invalidateAll = unit => Js.Promise.t(unit);
-type update('store, 'value) =
-  (~callback: (~store: 'store, ~key: string, ~value: 'value) => unit) =>
-  Js.Promise.t(unit);
-
-[@bs.deriving jsConverter]
-type cache('store, 'value) = {
-  write,
-  read: read('value),
-  invalidate,
-  invalidateAll,
-  update: update('store, 'value),
-};
-
-type cacheJs('store, 'value) = {
-  .
-  "write": write,
-  "read": read('value),
-  "invalidate": invalidate,
-  "invalidateAll": invalidateAll,
-  "update": update('store, 'value),
-};
+let unwrapFetchOptions = fetchOptions =>
+  switch (fetchOptions) {
+  | FetchObj(obj) => obj
+  | FetchFn(fn) => fn()
+  };
 
 [@bs.deriving abstract]
-type clientConfig('store, 'value) = {
+type clientOptions = {
   url: string,
   [@bs.optional]
-  cache: cacheJs('store, 'value),
-  [@bs.optional]
-  initialCache: 'store,
-  [@bs.optional]
   fetchOptions: Fetch.requestInit,
 };
 
-type client;
-
-[@bs.new] [@bs.module "urql"]
-external createClient: clientConfig('store, 'value) => client = "Client";
+[@bs.new] [@bs.module "urql"] external client: clientOptions => t = "Client";
 
 [@bs.send]
 external executeQuery:
-  (~client: client, ~query: UrqlQuery.urqlQuery, ~skipCache: bool) =>
-  Js.Promise.t('a) =
+  (~client: t, ~query: UrqlTypes.graphqlRequest) => Wonka.Types.sourceT('a) =
```

Comment:
This type param to `sourceT` here should be an `OperationResult` as defined here: https://github.com/FormidableLabs/urql/blob/774e09db8613d79e3e5cc6fca3d7f91526765f8e/src/types.ts#L44.